### PR TITLE
Add make-docs script that can be used to build multiple documentation sets

### PR DESCRIPTION
**Pull request by jdbaldry**
With the script served from the writers-toolkit documentation, it can be sourced by any repository and used everywhere.

The simplifications to the Makefile eliminate the need to update every repository when a new feature is added to the local webserver workflow.

The script encapsulates all the best practices we have established across the various `make docs` workflows, including:

1. Using `--init` flag to ensure that ^C works on MacOS.
1. Using `--platform linux/amd64` to ensure virtualization on MacOS.
1. Using `podman` instead of `docker` if the command is available.
1. Implementing a redirect from the project root to the mounted version.
1. Logging the webserver URLs for convenient browsing.

Additionally, this will let us have make-docs scripts in GEx products that automatically mount the relevant OSS content too.

### CODE REVIEW

**Comment by jdbaldry** [comment]
@osg-grafana This will let me bring the future improvements to all the projects with a single update. rather than multiple PRs.

@knylander-grafana One feature in particular this will propagate is the use of the `--platform` flag so you don't have to keep modifying the Makefile.

File: docs/sources/writing-guide/tooling-and-workflows/run-a-local-webserver/index.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+---
+title: Run a local documentation webserver with the make-docs script
+menuTitle: Run a local documentation webserver
+description: How to run a local documentation webserver with the make-docs script.
```
Comment (osg-grafana) [inline docs/sources/writing-guide/tooling-and-workflows/run-a-local-webserver/index.md:4]:
```suggestion
description: Run a local documentation web server with the make-docs script.
```

File: docs/sources/writing-guide/tooling-and-workflows/run-a-local-webserver/index.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+---
+
+# Run a local documentation webserver with the make-docs script
+
+The [`make-docs`](https://github.com/grafana/writers-toolkit/blob/main/scripts/make-docs) script can be used to mount multiple documentation sets in a single build.
+It is used by the `make docs` Make target in project repositories and can also be run separately.
```
Comment (osg-grafana) [inline docs/sources/writing-guide/tooling-and-workflows/run-a-local-webserver/index.md:10]:
```suggestion
To build multiple documentation sets within a single build, you can use the [`make-docs`](https://github.com/grafana/writers-toolkit/blob/main/scripts/make-docs) script, which you can also run on an individual documentation set.

> **Note:** The similar looking `make docs` Make target uses it within project repositories.
```

File: docs/sources/writing-guide/tooling-and-workflows/run-a-local-webserver/index.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+## Examples
+
+The following examples assume:
+
+1. Your working directory is the root of a project repository.
+1. You have a copy of the `make-docs` script in that working directory.
+1. The script is marked as executable.
```
Comment (osg-grafana) [inline docs/sources/writing-guide/tooling-and-workflows/run-a-local-webserver/index.md:18]:
```suggestion
1. The script is executable.
```

File: docs/sources/writing-guide/tooling-and-workflows/run-a-local-webserver/index.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+1. You have a checkout of each of the project repositories used in the examples.
```
Comment (osg-grafana) [inline docs/sources/writing-guide/tooling-and-workflows/run-a-local-webserver/index.md:19]:
```suggestion
1. You have a checkout of each of the project repositories that are used in the examples.
```

**Review by osg-grafana — APPROVED** [review]
Unblocking with some (hopefully) clarifying feedback; did not do a thorough review after my final comment because it is best to ship it. :)

**Comment by jdbaldry** [comment]
> Unblocking with some (hopefully) clarifying feedback; did not do a thorough review after my final comment because it is best to ship it. :)

Thanks for the review! All feedback is always welcome :)

**Review by chri2547 — APPROVED** [review]

**Pull request merged**